pkg/trtl/metrics/metrics.go

```diff
@@ -0,0 +1,239 @@
+package trtl
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/rs/zerolog/log"
+)
+
+var (
+	// Basic RPC Metrics
+	PmPuts       *prometheus.CounterVec   // count of trtl Puts per namespace
+	PmGets       *prometheus.CounterVec   // count of trtl Gets per namespace
+	PmDels       *prometheus.CounterVec   // count of trtl Deletes per namespace
+	PmIters      *prometheus.CounterVec   // count of trtl Iters per namespace
+	PmRPCLatency *prometheus.HistogramVec // the time it is taking for successful RPC calls to complete, labeled by RPC type, success, and failure
+	// PmObjects    *prometheus.CounterVec   // count of objects being managed by trtl, by namespace
+	// PmTombstones *prometheus.CounterVec   // count of tombstones per namespace; increases on delete, decrease on overwrite of tombstone
+
+	// Anti-Entropy Metrics
+	PmAESyncs         *prometheus.CounterVec   // count of anti entropy sessions per peer, per region, and by perspective (initiator/remote)
+	PmAESyncLatency   *prometheus.HistogramVec // duration of anti entropy sessions (initiator perspective), by peer
+	PmAEPhase1Latency *prometheus.HistogramVec // time phase 1 of anti-entropy is taking from the perspective of the initiator, by peer
+	PmAEPhase2Latency *prometheus.HistogramVec // time phase 2 of anti-entropy is taking from the perspective of the remote, by peer
+	PmAEVersions      *prometheus.HistogramVec // count of all observed versions, per peer and region
```

Comment:
potentially this makes more sense as a counter?

Reply:
I see what you're saying - but that's a tough one. On the one hand, we could count all the version vectors we send, which is a proxy for baseline cost; on the other we measure the size of the database and how it grows over time. For now (e.g. until we start monitoring in earnest to figure out what we're looking for) I think the histogram is the most understandable so I agree with your choice here.